### PR TITLE
[GCP] Fix unbounded return value

### PR DESCRIPTION
**Pull request by Michaelvll**
This is a bug in GCP node_provider, where an unnecessary variable is returned, which is not defined in one code path.

Reference: https://github.com/ray-project/ray/blob/547dd8c125b0e2007b1fb8960b514bd2c89e1963/python/ray/autoscaler/_private/gcp/node_provider.py#L183-L205

~~The parallel termination is adopted from: https://github.com/ray-project/ray/commit/46fc66390cb136a52afd6d70808f7c01eb31deb0~~

We revert the adoption of the parallel termination as that implementation is buggy with the following repro:
1. `sky launch -c test --cloud gcp --cpus 2 --num-nodes 8`
2. `sky stop test`, the following error shows up:
```
malloc(): unsorted double linked list corrupted
```

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
  - [x] `sky launch --num-nodes 8 --cloud gcp -c test-mn`; `sky stop test-mn`
  - [x] `sky launch --num-nodes 8 --cloud gcp -c test-mn`; `sky down -y test-mn`
- [x] All smoke tests: `pytest tests/test_smoke.py`

### CODE REVIEW

**Comment by concretevitamin** [comment]
Q: what problems did this create? 



> Reference: https://github.com/ray-project/ray/blob/547dd8c125b0e2007b1fb8960b514bd2c89e1963/python/ray/autoscaler/_private/gcp/node_provider.py#L183-L205

Not following this ref. It has newer code (parallel termination).

**Comment by Michaelvll** [comment]
> Q: what problems did this create?
> 
> > Reference: https://github.com/ray-project/ray/blob/547dd8c125b0e2007b1fb8960b514bd2c89e1963/python/ray/autoscaler/_private/gcp/node_provider.py#L183-L205
> 
> Not following this ref. It has newer code (parallel termination).

The problem is that the local variable `results` is used to return, which is not defined the following lines are triggered. 
https://github.com/ray-project/ray/blob/547dd8c125b0e2007b1fb8960b514bd2c89e1963/python/ray/autoscaler/_private/gcp/node_provider.py#L192-L197

The original code in ray is buggy and the returned `results` is not used, so it is safe to remove the return value.

Good catch for their parallel termination support. I am going to adopt the changes in our code in the this PR : )

**Comment by Michaelvll** [comment]
Hey @concretevitamin, please feel free to take another look. : )

File: sky/skylet/providers/gcp/node_provider.py [hunk 1]
```diff
@@ -336,9 +337,11 @@ def terminate_node(self, node_id: str):
                         f"Tried to delete the node with id {node_id} "
                         "but it was already gone."
                     )
+                    result = None
```
Comment (concretevitamin) [inline sky/skylet/providers/gcp/node_provider.py:340]: nit: would it be better to put it at L295 or something?

File: sky/skylet/providers/gcp/node_provider.py [hunk 2]
```diff
@@ -1,9 +1,10 @@
+import concurrent.futures
```
Comment (concretevitamin) [inline sky/skylet/providers/gcp/node_provider.py:1]: Is this needed?

**Review by concretevitamin — APPROVED** [review]
Sorry for missing this. LGTM.

File: sky/skylet/providers/gcp/node_provider.py (hunk 2 above)
Comment (Michaelvll) [reply sky/skylet/providers/gcp/node_provider.py]: Ahh, good catch! Removed. Thanks!

**Pull request merged**